src/cache/cache.ts

```diff
@@ -280,6 +280,20 @@ export function createCache(ctx: Context, bundleContext: IBundleContext): ICache
         }
       }
     }
+
+    // for any dependencies that have package.json
+    // if the package.json changes, then the "main" might change
+    // which might change the target.absPath
+    // so we can no longer trust the dependencies and have to look them up again
```

Comment:
This is the change I was least certain about.  There is code elsewhere that decides whether to call `restorePackage()` or to call `restoreModuleSafely()`.  I mostly left that code how it is, but then one of the tests fails when it modifies a `package.json` because of the issue described in this comment.  So I added this code here which resolves it, but I'm not sure whether this is the best place or not.